remote-drivers/lwip-ethernet-async/include/lwip-ethernet-async.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020, Data61
+ * Commonwealth Scientific and Industrial Research Organisation (CSIRO)
+ * ABN 41 687 119 230.
+ *
+ * This software may be distributed and modified according to the terms of
+ * the BSD 2-Clause license. Note that NO WARRANTY is provided.
+ * See "LICENSE_BSD2.txt" for details.
+ *
+ * @TAG(DATA61_BSD)
+ */
+
+#include <sel4/sel4.h>
+#include <platsupport/io.h>
+
+#define ENCODE_DMA_ADDRESS(buf) ({ \
```

Comment:
does this really have to be a macro or could it also become a static-inline function?